mne/decoding/base.py

```diff
@@ -83,25 +83,27 @@ def fit(self, X, y):
         if X.ndim != 2:
             raise ValueError('LinearModel only accepts 2-dimensional X, got '
                              '%s instead.' % (X.shape,))
-
         # fit the Model
         self.model.fit(X, y)
 
-        # computes the patterns
-        if (
-            not hasattr(self.model, 'coef_') or  # missing attribute
-            self.model.coef_.ndim > 2 or         # weird case
-            (self.model.coef_.size not in
-             self.model.coef_.shape)             # shape (n), (n, 1) or (1, n)
-        ):
+        # Computes patterns using Haufe's trick:
+        # A = Covariance_X . W . Precision_Y
 
-            raise ValueError('model needs a unidimensional coef_ attribute to '
-                             'compute the patterns')
-        self.filters_ = np.squeeze(self.model.coef_)
-        self.patterns_ = np.cov(X.T).dot(self.filters_)
+        inv_Y = 1.
+        X = X - X.mean(0, keepdims=True)
+        if np.squeeze(y).ndim > 1:
+            y = y - y.mean(0, keepdims=True)
+            inv_Y = np.linalg.pinv(np.cov(y.T))
+        self.patterns_ = np.cov(X.T).dot(self.filters_.T.dot(inv_Y)).T
 
         return self
 
+    @property
+    def filters_(self):
+        if not hasattr(self.model, 'coef_'):
+            raise ValueError('model does not have a `coef_` attribute.')
+        return np.squeeze(self.model.coef_)
```

Comment:
Docstring does not suggest that these can be 1D (but the squeeze can make it so)